nix-buffer.el

```diff
@@ -216,9 +216,10 @@ is removed."
   (interactive)
   (let* ((root (directory-file-name (or (buffer-file-name) default-directory)))
 	 (expr-dir (locate-dominating-file root "dir-locals.nix")))
-    (when expr-dir
-	 (let ((expr-file (f-expand "dir-locals.nix" expr-dir)))
-	   (nix-buffer--nix-build root expr-file)))))
+    (if expr-dir
+	(let ((expr-file (f-expand "dir-locals.nix" expr-dir)))
+	  (nix-buffer--nix-build root expr-file))
+      (error "No dir-locals.nix file found"))))
```

Comment:
Yeah I mean projectile uses find-file-hook too but it utilizes some caching and stuff:

https://github.com/bbatsov/projectile/blob/master/projectile.el#L3688

The error for me was just helpful debugging what was going on.  I can remove it though